File: 0000-0999/0242. Valid Anagram/Solution.cpp
/**
 * Author: Fbin
 * Link:https://leetcode.com/problems/valid-anagram/description/?envType=study-plan-v2&envId=top-interview-150
 * Created at: 30/04/2025
 */

#include <bits/stdc++.h>
using namespace std;

class Solution {
 public:
  bool isAnagram(string s, string t) {
    if (s.length() != t.length()) return false;

    int freq[26];
    for (int i = 0; i < s.length(); i++) {
      freq[s[i] - 'a']++;
      freq[t[i] - 'a']--;
    }

    for (int& f : freq) {
      if (f != 0) return false;
    }

    return true;
  }
};

File: 0000-0999/0242. Valid Anagram/Solution.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	freq := map[int]int{}
	for i := 0; i < len(s); i++ {
		freq[int(s[i])]++
		freq[int(t[i])]--
	}

	for _, f := range freq {
		if f != 0 {
			return false
		}
	}

	return true
}
